### PR TITLE
BlobStore updates

### PR DESCRIPTION
**Pull request by jahhulbert-ccri**
Rebased, two commits...testing now might have 1-2 more changes

Please look at the 2nd commit changes too.

2nd commit:
* Closeable interface
* Rename api to web 
* move packages
* code refac, testing

### CODE REVIEW

**Comment by jahhulbert-ccri** [comment]
probably should have some install doc

File: geomesa-blobstore/geomesa-blobstore-core/src/main/java/org/locationtech/geomesa/blob/core/GeoMesaBlobStore.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+     */
+    String put(File file, Map<String, String> params);
+
+    /**
+     * Adds a byte array to the blobstore, relying on user provided params for geo-indexing
+     * @param bytes to ingest, bypass FileHandlers to rely on client to set params
+     * @param params Map of parameters - see implementation for details
+     * @return Blob id as a string or null if put failed
+     */
+    String put(byte[] bytes, Map<String, String> params);
```
Comment (elahrvivaz) [inline geomesa-blobstore/geomesa-blobstore-core/src/main/java/org/locationtech/geomesa/blob/core/GeoMesaBlobStore.java]: might be nice to accept an input stream too... for another PR

File: geomesa-blobstore/geomesa-blobstore-core/src/main/java/org/locationtech/geomesa/blob/core/impl/AccumuloGeoMesaBlobStore.java [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                                    final String auths,
+                                    final Boolean useMock) throws IOException {
+        this(new HashMap<String, Serializable>() {{
+            put(AccumuloDataStoreParams.instanceIdParam().key, instanceId);
+            put(AccumuloDataStoreParams.tableNameParam().key, tableName);
+            put(AccumuloDataStoreParams.zookeepersParam().key, zookeepers);
+            put(AccumuloDataStoreParams.userParam().key, user);
+            put(AccumuloDataStoreParams.passwordParam().key, password);
+            put(AccumuloDataStoreParams.authsParam().key, auths);
+            put(AccumuloDataStoreParams.mockParam().key, useMock.toString());
```
Comment (elahrvivaz) [inline geomesa-blobstore/geomesa-blobstore-core/src/main/java/org/locationtech/geomesa/blob/core/impl/AccumuloGeoMesaBlobStore.java]: haha, this is very scala-y java

File: geomesa-blobstore/geomesa-blobstore-core/src/main/scala/org/locationtech/geomesa/blob/core/AccumuloBlobStore.scala [hunk 3]
```diff
@@ -89,83 +85,84 @@ class AccumuloBlobStore(ds: AccumuloDataStore) extends LazyLogging with BlobStor
...
-      while (fd.hasNext) {
-        fd.next()
-        fd.remove()
+      scanner.setRange(new Range(new Text(id)))
+
+      val iter = SelfClosingIterator(scanner)
+      if (iter.hasNext) {
+        val next = iter.next()
+        val ret = (next.getKey.getColumnQualifier.toString, next.getValue.get)
+        iter.close()
```
Comment (elahrvivaz) [inline geomesa-blobstore/geomesa-blobstore-core/src/main/scala/org/locationtech/geomesa/blob/core/AccumuloBlobStore.scala]: you've got some duplicate closing going on here, plus the selfClosing, plus the finally

**Comment by elahrvivaz** [comment]
looks good

File: geomesa-blobstore/geomesa-blobstore-core/src/main/java/org/locationtech/geomesa/blob/core/impl/AccumuloGeoMesaBlobStore.java (hunk 2 above)
Comment (jahhulbert-ccri) [inline geomesa-blobstore/geomesa-blobstore-core/src/main/java/org/locationtech/geomesa/blob/core/impl/AccumuloGeoMesaBlobStore.java]: hehehe yeah i prefer to call the constructor from the other constructor usually in Java so I went for it.

File: geomesa-blobstore/geomesa-blobstore-core/src/main/scala/org/locationtech/geomesa/blob/core/AccumuloBlobStore.scala (hunk 3 above)
Comment (jahhulbert-ccri) [inline geomesa-blobstore/geomesa-blobstore-core/src/main/scala/org/locationtech/geomesa/blob/core/AccumuloBlobStore.scala]: yeah...lemme fix that up

**Comment by elahrvivaz** [comment]
merging

**Comment by elahrvivaz** [comment]
951a3856806b389f9a49d969283c6d683de162aa
986d3848bd2b50504b8b86fd00a62de8ee4c8a82

**Pull request closed without merging**